projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts

```diff
@@ -27,8 +27,7 @@ export class DataLayerColorService {
     line.backgroundColor = color + '33'; // temporary dirty hack to set opacity. assumes color is in 6-character hex format.
   }
 
-  getAnnotationColor(annotation: any): string | undefined {
-    const color = annotation.backgroundColor;
+  getAnnotationColor(color: any): string | undefined {
     if (typeof color === 'string' && !color.includes('33')) {
       return color + '33';
     }
```

Comment:
I find this confusing that the `getAnnotationColor` method no longer gets the value from the annotation. Based on the name of the method, I would expect it to do something similar to `getColor`. It looks like this is now a utility method for adding transparency to an abritrary color code, so let's change its name to describe what it is doing.

There's also a bug in this method because `includes` will match a '33' anywhere in the string, not just at the end.

Reply:
right, @semicolin i will change it in the component, we don't need a service method for that add a '33' in color

Reply:
instead of includes we use endsWith() for checking whether '33' is present in end or not.
And function name changed from getAnnotationColor to addTransparencyAbritraryColor.